analyzer/analyzer.spicy

```diff
@@ -68,27 +68,28 @@ type Options = unit {
 # RFC-1533: DHCP Options.
 type Option = unit {
     code: uint8 &convert=OptionCode($$);
-    switch ( self.code ) {
-        OptionCode::PAD, OptionCode::END -> : void {}
+    switch (self.code) {
+        OptionCode::PAD,
+        OptionCode::END -> : void {}
         OptionCode::TIME_OFFSET -> {
             : skip uint8;
             time_offset: int32;
         }
         OptionCode::ROUTER -> {
             len: uint8;
-            routers: (addr &ipv4)[self.len/4] ; # `len` is always a multiple of 4.
+            routers: (addr &ipv4)[self.len / 4]; # `len` is always a multiple of 4.
```

Comment:
Independent of that change, I wonder if `&requires(self.len % 4 == 0)` above would make sense.

Reply:
Added. I also added a check `len>0` below.

I was also briefly wondering whether it would make sense to constrain to some maximum length, but then I recalled that DHCP was over UDP 😅.